parachain/pallets/inbound-queue/src/lib.rs

```diff
@@ -117,10 +137,9 @@ pub mod pallet {
 				}
 			})?;
 
-			let message_id = MessageId::new(envelope.account, envelope.nonce);
-			T::MessageDispatch::dispatch(envelope.source, message_id, &envelope.payload);
-
-			<LatestVerifiedBlockNumber<T>>::set(block_number);
+			// Reward relayer from the sovereign account of the destination parachain
+			let dest_account = envelope.dest.into_account_truncating();
+			T::Token::transfer(&who, &dest_account, T::Reward::get(), Preservation::Expendable)?;
```

Comment:
Seems a little confused. Is the transfer logic here correct or just some temp placeholder?  Will the impl here supposed to be a xcm call sent to asset chain?

Reply:
And also seems we remove the previous logic `<LatestVerifiedBlockNumber<T>>::set(block_number);` Is it safe to remove? I would assume we still need some scanner logic to find the unsynced messages and it is useful to narrow the scan scope down.

Reply:
> Seems a little confused. Is the transfer logic here correct or just some temp placeholder? Is impl supposed to be some xcm call sent to asset chain?

That's just for rewarding the relayer. We definitely still need to implement the routing of the XCM call.

Reply:
I temporarily removed the `LatestVerifiedBlockNumber` logic because the checkpoint logic in the relayer didn't work with multiple lanes (i.e each parachain has its lane and nonce).

So I am may add it back once I rewrite the relayer properly.